docs.cask.co/scripts/builder.py

```diff
@@ -0,0 +1,332 @@
+# License for the specific language governing permissions and limitations under
+# the License.
+
+# builder.py to create JSON file
+
+# version 0.4
+
+# JSON Creation (json-versions.js)
+# Creates a JSON file with timeline and formatting information from the data in a 
+# supplied configuration file
+# conf.py ("versions_data")
+
+import os
+import sys
+from datetime import datetime
+from optparse import OptionParser
+
+
+def parse_options():
+    """ Parses args options.
+    """
+
+    parser = OptionParser(
+        usage="%prog source target",
+        description="Builds a timeline and menu in Javascript format for a web page.")
+
+    (options, args) = parser.parse_args()
+    
+    if len(args) < 2:
+        parser.print_help()
+        print "A 'source' and a 'target' must be supplied."
+        sys.exit(1)
+                
+    return options, args, parser
+
+def _build_timeline():
+    """Takes a dictionary ('versions_data') from a global variable ('configuration') in a conf.py
+    and creates a timeline in a json file
+    
+    versions_data is a dictionary. 
+    
+    This modifies the "older", adding an extra element ('1') to flag the highest version(s) of the minor index.
+    To be backward-compatible with older documentation sets, the "included-in-menu" field moves to index 4:
+    
+    ['2.7.1', '2.7.1', '2015-02-05', '1', '0']
+    
+    "versions_data":
+    { "development": [
+        ['3.1.0-SNAPSHOT', '3.1.0'], 
+        ], 
+      "current": ['3.0.1', '3.0.0', '2015-05-07'], 
+      "older": [ 
+        ['3.0.0', '3.0.0', '2015-05-05'], 
+        ['2.8.0', '2.8.0', '2015-03-23'], 
+        ['2.7.1', '2.7.1', '2015-02-05'], 
+        ['2.6.3', '2.6.3', '2015-05-15'], 
+        ['2.6.2', '2.6.2', '2015-03-23'], 
+        ['2.6.1', '2.6.1', '2015-01-29'], 
+        ['2.6.0', '2.6.0', '2015-01-10'], 
+        ['2.5.2', '2.5.2', '2014-11-14'], 
+        ['2.5.1', '2.5.1', '2014-10-15'], 
+        ['2.5.0', '2.5.0', '2014-09-26'],
+        ],
+    },
+
+    timeline is a list of lists. The first item of each inner list is the indentation of the inner list, with 0 being
+    far-left, and increasing by 1. It is the same as the "older", but re-arranged into a hierarchy:
+
+    'timeline': [
+        ['0', '3.0.0', '2015-05-05', ' (43 days)'],
+        ['0', '3.0.1', '2015-05-07', ' (2 days)'],
+        ['0', '2.8.0', '2015-03-23', ' (46 days)'],
+        ['0', '2.7.1', '2015-02-05', ' (26 days)'],
+        ['0', '2.6.0', '2015-01-10', ' (106 days)'],
+        ['1', '2.6.1', '2015-01-29', ' (19 days)'],
+        ['1', '2.6.2', '2015-03-23', ' (53 days)'],
+        ['0', '2.5.0', '2014-09-26', ''],
+        ['1', '2.5.1', '2014-10-15', ' (19 days)'],
+        ['1', '2.5.2', '2014-11-14', ' (30 days)'],
+    ]
+    
+    """
+    global configuration
+
+    versions_data = configuration["versions_data"]
+    older = versions_data["older"]
+    rev_older = list(older)
+    rev_older.insert(0, versions_data["current"])
```

Comment:
Use single quotes anywhere where you aren't using string interpolation... `versions_data = configuration['versions_data']`, etc